cms/cms_menus.py

```diff
@@ -137,23 +145,27 @@ def get_menu_node_for_page(renderer, page, language):
     if exts:
         attr['navigation_extenders'] = exts
 
-    translation = page.get_title_obj(language, fallback=True)
-
-    # Do we have a redirectURL?
-    attr['redirect_url'] = translation.redirect  # save redirect URL if any
-
-    # Now finally, build the NavigationNode object and return it.
-    ret_node = CMSNavigationNode(
-        translation.menu_title or translation.title,
-        url='',
-        id=page.pk,
-        parent_id=parent_id,
-        attr=attr,
-        visible=page.in_navigation,
-        path=translation.path or translation.slug,
-        language=(translation.language if translation.language != language else None),
-    )
-    return ret_node
+    for lang in [language] + fallbacks:
+        translation = page.title_cache[lang]
```

Comment:
possible ``KeyError``?

Reply:
The title_cache is populated with ``EmptyTitle`` instances for all fallbacks, these instances are considered ``False`` when evaluated below.